src/pydvl/parallel/backends/joblib.py

```diff
@@ -19,50 +20,72 @@
-
-    ??? Example
+    !!! Example
         ``` python
-        import joblib
-        from pydvl.parallel import init_paralle_backend, ParallelConfig
-        with joblib.parallel_config(verbose=100):
-            config = ParallelConfig(backend="joblib")
-            parallel_backend = init_parallel_backend(config)
+        from pydvl.parallel import JoblibParallelBackend
+        parallel_backend = JoblibParallelBackend()
         ```
-
-    Args:
-        config: instance of [ParallelConfig][pydvl.utils.config.ParallelConfig]
-            with cluster address, number of cpus, etc.
     """
 
-    def __init__(self, config: ParallelConfig):
+    @deprecated(
+        target=True,
+        args_mapping={"config": None},
+        deprecated_in="0.9.0",
+        remove_in="0.10.0",
+    )
+    def __init__(self, config: Optional[ParallelConfig] = None) -> None:
+        n_jobs: Optional[int] = None
+        if config is not None:
+            n_jobs = config.n_cpus_local
         self.config = {
-            "n_jobs": config.n_cpus_local,
+            "n_jobs": n_jobs,
         }
 
     @classmethod
     def executor(
         cls,
-        max_workers: int | None = None,
-        config: ParallelConfig = ParallelConfig(),
-        cancel_futures: CancellationPolicy = CancellationPolicy.NONE,
+        max_workers: Optional[int] = None,
+        *,
+        config: Optional[ParallelConfig] = None,
+        cancel_futures: Union[CancellationPolicy, bool] = CancellationPolicy.NONE,
     ) -> Executor:
+        """Returns a futures executor for the parallel backend.
+
+        !!! Example
+            ``` python
+            from pydvl.parallel import JoblibParallelBackend
+            parallel_backend = JoblibParallelBackend()
+            with parallel_backend.executor() as executor:
+                executor.submit(...)
+            ```
+
+        Args:
+            max_workers: Maximum number of parallel workers.
+            config: (**DEPRECATED**) Object configuring parallel computation,
+                with cluster address, number of cpus, etc.
+            cancel_futures: Policy to use when cancelling futures
+                after exiting an Executor.
+
+        Returns:
+            Instance of [_ReusablePoolExecutor][joblib.externals.loky.reusable_executor._ReusablePoolExecutor].
+        """
+        if config is not None:
+            warnings.warn(
+                "The `JoblibParallelBackend` uses deprecated arguments: "
+                "`config`. They were deprecated since v0.9.0 "
+                "and will be removed in v0.10.0.",
+                FutureWarning,
+            )
+
         if cancel_futures not in (CancellationPolicy.NONE, False):
             warnings.warn(
-                "Cancellation of futures is not supported by the joblib backend"
+                "Cancellation of futures is not supported by the joblib backend",
             )
-        return cast(Executor, get_reusable_executor(max_workers=max_workers))
```

Comment:
I think that cast should only be used sporadically because it's less readable than the alternative, at least for me. I will put it back the way it was for the sake of consistency though.

Reply:
Quite the opposite, I find cast rather more readable than the alternative. I had to look twice at those lines because of the unexpected extra variable, whereas cast clearly conveys the intention.

Reply:
I have changed it back.